frontend/src/components/Metrics/MetricsStep/DeploymentFrequencySettings/SingleSelection/index.tsx

```diff
@@ -21,21 +22,30 @@ export const SingleSelection = ({
   id,
   errorMessage,
   onGetSteps,
+  step,
   onUpDatePipeline,
   onClearErrorMessage,
 }: Props) => {
   const labelId = `single-selection-${label.toLowerCase().replace(' ', '-')}`
+  const [selectedOptions, setSelectedOptions] = useState(value)
 
   const handleChange = (event: SelectChangeEvent) => {
-    const value = event.target.value
+    const newValue = event.target.value
+    setSelectedOptions(newValue)
     if (onGetSteps) {
-      onGetSteps(value)
       onUpDatePipeline(id, 'Step', '')
+      onGetSteps(newValue)
     }
-    onUpDatePipeline(id, label, value)
+    onUpDatePipeline(id, label, newValue)
     !!errorMessage && onClearErrorMessage(id, camelCase(label))
   }
 
+  useEffect(() => {
+    if (onGetSteps && selectedOptions !== '' && step === '') {
```

Comment:
u can use !step replace step ==== '', also selectOptions

Reply:
I have refactored it